sycl/doc/GetStartedGuide.md

```diff
@@ -639,8 +639,10 @@ which contains all the symbols required.
 
 ## C++ standard
 
-* DPC++ runtime and headers require C++14 at least.
-* DPC++ compiler is building apps as C++17 apps by default.
+* DPC++ headers require C++17 at least.
+* DPC++ runtime requires C++14 at least.
```

Comment:
According to my understanding, DPC++ runtime uses DPC headers, which means if C++ version used in DPC++ header is higher than C++ used to build DPC++ runtime, it might not compile.

In order words DPC++ headers C++ version must be <= DPC++ runtime C++ version.

Reply:
I agree. If we want uplift we should uplift both having remaining issues at #3447 resolved.

Reply:
С++17 code in headers is not used in the library and is guarded by macro `#if __cplusplus > 201402L`, so the current combination of headers (which have some C++17 code) and library (C++14) works.

Reply:
I think you are correct, however it's sort of library internal knowledge, raising both looks more consistent.

Reply:
> С++17 code in headers is not used in the library and is guarded by macro `#if __cplusplus > 201402L`, so the current combination of headers (which have some C++17 code) and library (C++14) works.

That means that headers require C++14, not C++17.

Reply:
@bader, do you mean that adding C++17 code to the headers is some kind of hack now, and in this case the headers officially still require the same C++14 standard?

Reply:
> do you mean that adding C++17 code to the headers is some kind of hack now

No at all. Making use of C++17 features in DPC++ headers is not a hack.

> in this case the headers officially still require the same C++14 standard

DPC++ runtime is compiled in C++14 mode, which headers headers must be compatible with C++14 mode.

Reply:
Ok, I see. In this case I can leave this information w/o any changes and it should be updated in https://github.com/intel/llvm/pull/3447